jboss/container/wildfly/launch/access-log-valve/added/launch/access_log_valve.sh

```diff
@@ -0,0 +1,246 @@
+#!/bin/sh
+
+# Access Log Valve configuration script
+# Usage:
+#   it is disabled by default, to disable it set the following variable to true:
+#   ENABLE_ACCESS_LOG
+#
+# Default pattern used across all products:
+#   %h %l %u %t %{X-Forwarded-Host}i "%r" %s %b
+#          eap7 %{i,X-Forwarded-Host}
+#
+# Where:
+#   %h: Remote host name
+#   %l: Remote logical username from identd (always returns '-')
+#   %u: Remote user that was authenticated
+#   %t: Date and time, in Common Log Format format
+#   %{X-Forwarded-Host}: for X-Forwarded-Host incoming headers
+#   %r: First line of the request, generally something like this: "GET /index.jsf HTTP/1.1"
+#   %s: HTTP status code of the response
+#   %b: Bytes sent, excluding HTTP headers, or '-' if no bytes were sent
+#
+# Example of configuration that will be added on standalone-openshift.xml for eap6
+#   <valve name="accessLog" module="org.jboss.openshift" class-name="org.jboss.openshift.valves.StdoutAccessLogValve">
+#       <param param-name="pattern" param-value="%h %l %u %t %{X-Forwarded-Host}i "%r" %s %b" />
+#   </valve>
+#
+# Example for eap7.x
+#   <access-log use-server-log="true" pattern="%h %l %u %t %{i,X-Forwarded-Host} "%r" %s %b"/>
+#
+# This script will be executed during container startup
+
+source $JBOSS_HOME/bin/launch/logging.sh
+
+function configure() {
+    configure_access_log_valve
+  configure_access_log_handler
+}
+
+function configure_access_log_valve() {
+
+  local mode
+  getConfigurationMode "<!-- ##ACCESS_LOG_VALVE## -->" "mode"
+
+  if [ "${ENABLE_ACCESS_LOG^^}" == "TRUE" ]; then
+
+    if [ "${mode}" == "xml" ]; then
+      local pattern=$(getPattern "add-xml")
+      local valve="<access-log use-server-log=\"true\" pattern=\"${pattern}\"/>"
+      sed -i "s|<!-- ##ACCESS_LOG_VALVE## -->|${valve}|" $CONFIG_FILE
+    else
+            # A lot of XPath here since we need to do more advanced stuff than CLI allows us to...
+
+      # Check there is an Undertow subsystem
+      local subsystemRet
+      local xpath="\"//*[local-name()='subsystem' and starts-with(namespace-uri(), 'urn:jboss:domain:undertow:')]\""
+      testXpathExpression "${xpath}" "subsystemRet"
+      if [ "${subsystemRet}" -ne 0 ]; then
+        echo "You have set ENABLE_ACCESS_LOG=true to add the access-log valve. Fix your configuration to contain the undertow subsystem for this to happen." >> ${CLI_SCRIPT_ERROR_FILE}
+        exit 1
```

Comment:
I think we should not exit here. It will exit completely, right? If there are errors in the CLI_SCRIPT_ERROR_FILE they will be processed by openshift-common.sh then we will exit.

Reply:
You are right, I had this problem somewhere else